tracing/providers/opentelemetry/src/main/java/io/helidon/tracing/providers/opentelemetry/MutableOpenTelemetryBaggage.java

```diff
@@ -65,9 +89,33 @@ public BaggageBuilder toBuilder() {
     }
 
     void baggage(String key, String value) {
+        Objects.requireNonNull(key, "baggage key cannot be null");
+        Objects.requireNonNull(value, "baggage value cannot be null");
         values.put(key, new HBaggageEntry(value, new HBaggageEntryMetadata("")));
     }
 
+    @Override
+    public Optional<String> get(String key) {
+        return Optional.ofNullable(values.get(key).getValue());
+    }
+
+    @Override
+    public Set<String> keys() {
+        return Collections.unmodifiableSet(values.keySet());
```

Comment:
Depending on the frequency of the call, consider storing an unmodifiable `Set` as an instance variable and returning it directly.

Reply:
The `values` map can grow at any time, so the key set will not be stable. A `Set` instance field would need to be updated whenever baggage was added or a dirty bit set and the `Set` reconstituted lazily if dirty upon access. My gut feel is that `keys` will be infrequently invoked, at least seldom enough for optimizing this to be premature given the map's changeability.

Reply:
Fair. (Do note that `Collections#unmodifiable...` methods account for changing values "underneath"; that's why they're called `unmodifiable` and not, say, `immutable`.)

Reply:
Exactly. We want the caller to see the accurate key set without being able to mess with its contents.